src/main/java/carpet/script/utils/ShapesRenderer.java

```diff
@@ -386,7 +615,165 @@ public void renderLines(PoseStack matrices, Tesselator tessellator, BufferBuilde
             );
         }
     }
+    public static class RenderedPolyface extends RenderedShape<ShapeDispatcher.Polyface>
+    {
+        public RenderedPolyface(Minecraft client, ShapeDispatcher.ExpiringShape shape)
+        {
+            super(client, (ShapeDispatcher.Polyface)shape);
+        }
+        @Override
+        public void renderFaces(Tesselator tessellator, BufferBuilder bufferBuilder, double cx, double cy, double cz, float partialTick)
+        {       
+            if(shape.fa==0){return;}
+            if (shape.needf3b && !client.getEntityRenderDispatcher().shouldRenderHitBoxes()){
+                return;
+            }
+            if (shape.doublesided)
+                RenderSystem.disableCull();
+            else
+                RenderSystem.enableCull();
+            
+            /* 
+            RenderSystem.depthMask(false);
+            RenderSystem.depthFunc(516);
+            RenderSystem.enableBlend();
+            RenderSystem.blendFunc(GlStateManager.SourceFactor.ONE_MINUS_DST_ALPHA, GlStateManager.DestFactor.DST_ALPHA);
+             */
+            bufferBuilder.begin(shape.mode, DefaultVertexFormat.POSITION_COLOR);
+                for(int i=0;i<shape.vertex_list.size();i++){
+                    Vec3 vec=shape.vertex_list.get(i);
+                    if(shape.relative.get(i)){
+                        vec=shape.relativiseRender(client.level, vec, partialTick);
+                    }
+                    bufferBuilder.vertex(vec.x()-cx, vec.y()-cy, vec.z()-cz).color(shape.fr, shape.fg, shape.fb, shape.fa).endVertex();
+                }
+                tessellator.end();
 
+            /*
+                RenderSystem.depthFunc(515);
```

Comment:
there is still a lot of commented code here and there indicating you are still testing something. Please cleanup :)

Reply:
ok XD